python/gtsam/examples/TranslationAveragingExample.py

```diff
@@ -0,0 +1,112 @@
+                poses[j].translation() - poses[i].translation()))
+            translation_directions.append(gtsam.BinaryMeasurementUnit3(
+                i, j, i_Z_j, gtsam.noiseModel.Isotropic.Sigma(3, 0.01)))
+    # Add the last two rotations.
+    rotations.insert(len(poses) - 1, poses[-1].rotation())
+    rotations.insert(len(poses) - 2, poses[-2].rotation())
+    return (rotations, translation_directions)
+
+
+def estimate_poses_given_rot(measurements: gtsam.BinaryMeasurementsUnit3,
+                             rotations: gtsam.Values):
+    """Estimate poses given normalized translation directions and rotations between nodes.
+
+    Arguments:
+        measurements {BinaryMeasurementsUnit3}- List of translation direction from the first node to
+        the second node in the coordinate frame of the first node.
+        rotations {Values} -- Estimated rotations
+
+    Returns:
+        Values -- Estimated poses.
+    """
+
+    # Some hyperparameters.
+    max_1dsfm_projection_directions = 50
+    outlier_weight_threshold = 0.1
+
+    # Convert the translation directions to global frame using the rotations.
+    w_measurements = gtsam.BinaryMeasurementsUnit3()
+    for measurement in measurements:
+        w_measurements.append(gtsam.BinaryMeasurementUnit3(measurement.key1(), measurement.key2(), gtsam.Unit3(
+            rotations.atRot3(measurement.key1()).rotate(measurement.measured().point3())), measurement.noiseModel()))
+
+    # Indices of measurements that are to be used as projection directions. These are randomly chosen.
+    indices = np.random.choice(len(w_measurements), min(
+        max_1dsfm_projection_directions, len(w_measurements)), replace=False)
+    # Sample projection directions from the measurements.
+    projection_directions = [w_measurements[idx].measured() for idx in indices]
+
+    outlier_weights = []
+    # Find the outlier weights for each direction using MFAS.
+    for direction in projection_directions:
+        algorithm = gtsam.MFAS(w_measurements, direction)
+        outlier_weights.append(algorithm.computeOutlierWeights())
+
+    # Compute average of outlier weights.
+    avg_outlier_weights = {}
+    for outlier_weight_dict in outlier_weights:
+        for k, v in outlier_weight_dict.items():
+            if k in avg_outlier_weights:
+                avg_outlier_weights[k] += v / len(outlier_weights)
+            else:
+                avg_outlier_weights[k] = v / len(outlier_weights)
+
+    # Remove measurements that have weight greater than threshold.
+    inlier_measurements = gtsam.BinaryMeasurementsUnit3()
+    [inlier_measurements.append(m) for m in w_measurements if avg_outlier_weights[(
+        m.key1(), m.key2())] < outlier_weight_threshold]
+
+    # Run the optimizer to obtain translations for normalized directions.
+    translations = gtsam.TranslationRecovery(inlier_measurements).run()
+
+    poses = gtsam.Values()
+    for key in rotations.keys():
```

Comment:
can we rename `key` to something more descriptive?

Reply:
Not sure what to rename it, any suggestions?

Reply:
didn't quite follow this part, what does each key represent?

Reply:
maybe we could rename it as `wRc`?

Reply:
[Key](https://github.com/borglab/gtsam/blob/ec04369c8870754c058714221f5b36737b3db770/gtsam/inference/Key.h) is a type in GTSAM. If I am not wrong, it is equivalent to a size_t (like an unsigned integer). Translation Averaging, like much of other gtsam code, uses Keys to represent Camera IDs. gtsam.Values is used to store the rotations, and it supports lookup by the key of the camera. 
I don't think calling it wRc would be correct, since it is not a rotation. Maybe we could call it camera_id in external code, but doing so here would hide the fact that it is actually a gtsam Key.

Reply:
The term keypair used at another place in the code is a pair of keys. The pybind wrapper allows us to treat it like a tuple of ints in python.